File: practica2_820574_839304/java/Mul.java
/*
* Fichero: Mul.java
* Autores: Alicia Lázaro Huerta y Manel Jordá Puig Rubio
* Fecha: 27/02/2023
*/
import java.util.Stack;

public class Mul extends Instruccion{
    Mul(){
        this.nombre = "mul";
    }
    public void ejecutar(Stack<Integer> pila, int[] pc){
        int a, b;
        a = pila.pop();
        b = pila.pop();

        pila.push(a * b);
        pc[0]++;
    }
}

File: practica2_820574_839304/c++/mul.cc
#include "mul.h"

Mul::Mul() {
    nombre = "mul";
}

void Mul::ejecutar(PilaInts &pila, int &pc) {
    int a, b;
    a = pila.cima();
    pila.extraer();
    b = pila.cima();
    pila.extraer();
    pila.insertar(a * b);
    pc++;
}

string Mul::nombrar() {
    return nombre;
}